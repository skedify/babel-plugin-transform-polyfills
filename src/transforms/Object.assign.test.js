describe('Object.assign', () => {
  it('should replace Object.assign with a polyfill', () => {
    const result = compile(
      `
      const c = Object.assign({a:1}, {b:2})
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const e = Object.assign({a:1}, {b:Object.assign({c:3}, {d:4})})
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const c = Object.keys(Object.assign({a:1}, {b:2}))
    `
    )

    expect(result).toMatchSnapshot()
  })
})
