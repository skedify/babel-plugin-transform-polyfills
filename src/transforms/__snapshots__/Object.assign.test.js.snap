// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Object.assign should not import polyfill more than once 1`] = `
"import _Object_assign from \\"babel-plugin-transform-polyfills/polyfills/Object/assign\\";

const e = _Object_assign({ a: 1 }, { b: _Object_assign({ c: 3 }, { d: 4 }) });"
`;

exports[`Object.assign should replace Object.assign with a polyfill 1`] = `
"import _Object_assign from \\"babel-plugin-transform-polyfills/polyfills/Object/assign\\";

const c = _Object_assign({ a: 1 }, { b: 2 });"
`;

exports[`Object.assign should work when part of a call chain 1`] = `
"import _Object_assign from \\"babel-plugin-transform-polyfills/polyfills/Object/assign\\";

const c = Object.keys(_Object_assign({ a: 1 }, { b: 2 }));"
`;

exports[`Object.assign should work when using apply 1`] = `
"import _Object_assign from \\"babel-plugin-transform-polyfills/polyfills/Object/assign\\";

const c = _Object_assign.apply(null, [{ a: 1 }, { b: 2 }]);"
`;

exports[`Object.assign should work when using call 1`] = `
"import _Object_assign from \\"babel-plugin-transform-polyfills/polyfills/Object/assign\\";

const c = _Object_assign.call(null, { a: 1 }, { b: 2 });"
`;
